app/src/main/java/com/example/sunshine/helpers/WTLocationClient.java

```diff
@@ -0,0 +1,116 @@
+package com.example.sunshine.helpers;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+
+import com.example.sunshine.context.WTApplication;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * @author jiecongwang
+ */
+public class WTLocationClient implements LocationListener {
+
+    private final LocationManager mLocationManager;
+    private final Context mContext;
+    private final ArrayList<WTLocationListener> mListeners ;
+    private final static int DURATION =1000;
+
+
+    private static class SingletonHolder {
+        private final static WTLocationClient INSTANCE = new WTLocationClient();
+    }
+
+    public static WTLocationClient getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+
+
+    private WTLocationClient() {
+        mListeners = Lists.newArrayList();
+        mContext = WTApplication.getInstance().getApplicationContext();
+        mLocationManager = (LocationManager)mContext.getSystemService(Context.LOCATION_SERVICE);
+        List<String> providers = mLocationManager.getProviders(true);
```

Comment:
getLastKnowLocation will return the lastest location, If it is not important to distinguish the different location source, because we only care about the location result. May be there is no location for GPS indoor, but we use can use wifi network